pkg/csi/service/vanilla/controller.go

```diff
@@ -829,6 +851,20 @@ func isVsan67u3Release(ctx context.Context, c *controller) (bool, error) {
 	return false, nil
 }
 
+// initVolumeMigrationService is a helper method to initialize volumeMigrationService in controller
+func initVolumeMigrationService(ctx context.Context, c *controller) (migration.VolumeMigrationService, error) {
```

Comment:
I still dont see the point of this function. Why can't callers directly call `migration.GetVolumeMigrationService`?

In controller, the error handling can wrap the error before returning. 

Another point, `initVolumeMigrationService` doesn't need to return anything. This function is assigning the global variable `volumeMigrationService` within its scope. What's the point of assigning it to a global variable within the function and then returning a value that's assigning the same variable?

Reply:
I am fine with avoiding this function all together. But it conflicts with this review comment: https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/364#discussion_r491138029
@divyenpatel Do you still think we need `initVolumeMigrationService` to wrap the VMS instantiation ?

Reply:
Assignment to global var and then returning the global var again is removed from`initVolumeMigrationService`